Packs/Illumio/Integrations/IllumioCore/IllumioCore_test.py

```diff
@@ -38,3 +55,224 @@ def test_test_module(requests_mock, mock_client):
+     "{} is an invalid value for port. Value must be in 1 to 65535.".format(-300)),
+    ({"port": 8443, "protocol": "tcp", 'start_time': '2022-07-17T12:58:33.528Z',
+      'end_time': "2022-07-18T12:58:33.529Z", 'policy_decisions': ['dummy']},
+     "Invalid value for {}. Possible comma separated values are {}.".format('policy_decisions',
+                                                                            VALID_POLICY_DECISIONS)),
+    ({"port": 8443, "protocol": "dummy", 'start_time': '2022-07-17T12:58:33.528Z',
+      'end_time': "2022-07-18T12:58:33.529Z", 'policy_decisions': ['potentially_blocked']},
+     "{} is an invalid value for {}. Possible values are: {}.".format('dummy', 'protocol', VALID_PROTOCOLS))
+])
+def test_illumio_traffic_analysis_command_for_invalid_arguments(args, err_msg, mock_client):
+    """
+    Test case scenario for execution of illumio-traffic-analysis-command when invalid argument provided.
+
+    Given:
+        - command arguments for illumio_traffic_analysis_command
+    When:
+        - Calling `illumio_traffic_analysis_command` function
+    Then:
+        - Returns a valid error message
+    """
+    from IllumioCore import illumio_traffic_analysis_command
+    with pytest.raises((ValueError, IllumioException, InvalidPortException, InvalidSingleSelectException,
+                        InvalidMultiSelectException)) as err:
+        illumio_traffic_analysis_command(mock_client, args)
+    assert str(err.value) == err_msg
+
+
+def test_illumio_traffic_analysis_success(requests_mock, mock_client):
+    """
+    Test case scenario for successful execution of illumio_traffic_analysis_command function.
+
+    Given:
+        - command arguments for illumio_traffic_analysis
+    When:
+        - Calling `illumio_traffic_analysis_Command` function
+    Then:
+        - Returns a valid output
+    """
+    from IllumioCore import illumio_traffic_analysis_command
+
+    args = {'port': 8443, 'protocol': "tcp", 'policy_decisions': 'potentially_blocked',
+            'start_time': "2022-07-17T12:58:33.528Z", 'end_time': "2022-07-18T12:58:33.529Z"}
+
+    json_data = util_load_json(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                            "test_data/traffic_analysis_success_response.json"))
+
+    with open(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                           "./test_data/traffic_analysis_success_hr.md")) as file:
+        hr_output = file.read()
+
+    json_data_get = util_load_json(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                                "test_data/traffic_analysis_get_response.json"))
+
+    json_download_data = util_load_json(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                                     "test_data/traffic_analysis_download_data.json"))
+
+    requests_mock.post("https://127.0.0.1:8443/api/v2/orgs/1/traffic_flows/async_queries",
```

Comment:
Shift these into a fixture, and just use the path for the pattern match.

Reply:
Yes

Reply:
Can we make a constant for the base_url & then use it instead of fixture?

Reply:
There's no reason to have the base URL at all - the fixture is to externalize the requests mock setup. Typically I would use a single auto-use fixture for this, but since this is all condensed into one module it probably makes more sense to have one fixture for each endpoint.

Reply:
@dsommerville-illumio, if we move all these `requests_mock `calls into the fixture then in the test function we would only have function call & assert. Also, this would require us to rewrite all the test cases and need to write fixtures for each endpoint call as well.
So do you want us to spend time on this & rewrite all the test cases?

Reply:
I'm not sure what you mean. The point of fixtures is to break out duplicated functionality, for example in the virtual service tests you have

```
    create_virtual_service_expected_resp = util_load_json(
        os.path.join(os.path.dirname(os.path.realpath(__file__)),
                     "test_data/create_virtual_service_success_response.json"))

```

in a couple of functions. This could be shifted to a fixture:

```
@pytest.fixture(scope='module')
def vs_success_response():
    return util_load_json(
        os.path.join(os.path.dirname(os.path.realpath(__file__)),
                     "test_data/create_virtual_service_success_response.json"))

def test_vs_example(vs_success_response):
    ...
```

Same with the metadata files. The requests mocks are slightly different, but often an autouse fixture is a good starting point when you have a generic response for an endpoint.

For traffic analysis, this should be simplified - there's really no point mocking all the specific endpoints. Just mock the function response itself. You're not testing the `illumio` library, you're testing the logic in the SOAR functions.

Reply:
@dsommerville-illumio, we've used monkey-patching for mocking the function call directly, so we've created an extra .`py ` file for storing the object response from the SDK in `test_data ` directory. We need to confirm whether we can have a .`py` file in test_data. If not, we'll have to move the responses directly into the test_file. Also, created fixtures according to the suggestions.